pilot/pkg/model/push_context.go

```diff
@@ -2177,6 +2177,24 @@ func (ps *PushContext) EnvoyFilters(proxy *Proxy) *EnvoyFilterWrapper {
 		matchedEnvoyFilters = append(matchedEnvoyFilters, matched...)
 	}
 
+	sort.Slice(matchedEnvoyFilters, func(i, j int) bool {
```

Comment:
you can use sort.SliceStable, then you donot need to check if ifilter.Priority != jfilter.Priority

 or if ifilter.Namespace != jfilter.Namespace

Reply:
otherwise we need to check if priority is set or not first - So I think this should be fine?